src/tools/MonarchPeasantSample/AppState.cpp

```diff
@@ -0,0 +1,113 @@
+#include "pch.h"
+#include <conio.h>
+#include "Monarch.h"
+#include "Peasant.h"
+#include "AppState.h"
+#include "../../types/inc/utils.hpp"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+using namespace ::Microsoft::Console;
+
+void AppState::_setupConsole()
+{
+    hOutput = GetStdHandle(STD_OUTPUT_HANDLE);
+    hInput = GetStdHandle(STD_INPUT_HANDLE);
+    DWORD dwMode = 0;
+    GetConsoleMode(hOutput, &dwMode);
+    dwMode |= ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+    SetConsoleMode(hOutput, dwMode);
+}
+
+void AppState::initializeState()
+{
+    // Initialize the console handles
+    _setupConsole();
+
+    // Set up WinRT
+    init_apartment();
+}
+
+bool AppState::areWeTheKing(const bool logPIDs)
+{
+    auto kingPID = monarch.GetPID();
+    auto ourPID = GetCurrentProcessId();
+    if (logPIDs)
+    {
+        if (ourPID == kingPID)
+        {
+            printf(fmt::format("We're the\x1b[33m king\x1b[m - our PID is {}\n", ourPID).c_str());
+        }
+        else
+        {
+            printf(fmt::format("We're a lowly peasant - the king is {}\n", kingPID).c_str());
+        }
+    }
+    return (ourPID == kingPID);
+}
+
+void AppState::remindKingWhoTheyAre(const winrt::MonarchPeasantSample::IPeasant& iPeasant)
+{
+    winrt::com_ptr<MonarchPeasantSample::implementation::Monarch> monarchImpl;
+    monarchImpl.copy_from(winrt::get_self<MonarchPeasantSample::implementation::Monarch>(monarch));
+    if (monarchImpl)
+    {
+        auto ourID = iPeasant.GetID();
+        monarchImpl->SetSelfID(ourID);
+        monarchImpl->AddPeasant(iPeasant);
+        printf("The king is peasant #%lld\n", ourID);
+    }
+    else
+    {
+        // printf("Shoot, we wanted to be able to get the impl here but couldnt\n");

```

Comment:
Shouldn't we actually print these things out? There's a few of these commented out printfs in here.